src/goose/cli/session.py

```diff
@@ -156,10 +156,10 @@ def run(self) -> None:
                 self.reply()  # Process the user message.
             except KeyboardInterrupt:
                 self.interrupt_reply()
-            except Exception:
+            except FailedToGenerateMessageError:
```

Comment:
Left a related message in the other PR, I think a more generic exception catch does make sense here
https://github.com/square/exchange/pull/30#discussion_r1746395686